### PR TITLE
[master] [IMP] sale, pos_sale: facilitate inheritance on sale report

### PR DESCRIPTION
**Pull request by celm1990**
**Description of the issue/feature this PR addresses:**
modules installed: `sale, pos_sale, module1`(depends of `sale`), `module 2`(depends of `point_of_sale `and `pos_sale`)
Currently is imposible add custom field to `pos.order `or `pos.order.line` and reflect field value into `sale.report`.
If custom field is added on `sale.order` or` sale.order.line` can inherit function `_query`, [example here](https://github.com/odoo/odoo/blob/master/addons/sale_margin/report/sale_report.py#L13) and add field to report easily. But if add same field to model `pos.order` **field value always is `NULL`** [for this function](https://github.com/odoo/odoo/blob/13.0/addons/pos_sale/report/sale_report.py#L70)

**Current behavior before PR:**
Add a field(new_field_id ) to `sale.order `and `pos.order` into two modules

`MODULE 1`(depends sale) working fine
```
class SaleOrder(models.Model):
    _inherit = 'sale.order'

    new_field_id = fields.Many2one('any.model')


class SaleReport(models.Model):
    _inherit = 'sale.report'

    new_field_id = fields.Many2one('any.model')

    def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
        fields['new_field_id '] = ", s.new_field_id"
        groupby += " , s.new_field_id"
        return super(SaleReport, self)._query(with_clause, fields, groupby, from_clause)
```

`MODULE 2`(depends `point_of_sale`, `pos_sale`, optional can add dependency for `module1 `but is not solution)

```
class PosOrder(models.Model):
    _inherit = 'pos.order'

    new_field_id = fields.Many2one('any.model')

```

For inherit sale.report i try this:
replace NULL(generate by https://github.com/odoo/odoo/blob/13.0/addons/pos_sale/report/sale_report.py#L70) after call super:

```
class SaleReport(models.Model):
    _inherit = 'sale.report'

    new_field_id = fields.Many2one('any.model')

    def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
        res =super(SaleReport, self)._query(with_clause, fields, groupby, from_clause)
        res = res .replace(", NULL AS new_field_id ", ", pos.new_field_id ")
        return res 
```

This raise error: `Field must appear in the GROUP BY clause or be used in an aggregate function`  So is imposibble add field to `GROUP BY` sentence and show value for field  new_field_id to sale.report

**Desired behavior after PR is merged:**

Can inherit easy and add field to `GROUP BY` as:

```
class SaleReport(models.Model):
    _inherit = "sale.report"

   new_field_id = fields.Many2one('any.model')

    def _select_pos(self, fields={}):
        select_ = super(SaleReport, self)._select_pos(fields=fields)
        select_ = select_ .replace(", NULL AS new_field_id ", ", pos.new_field_id ")
        return select_

    def _group_by_pos(self, groupby=''):
        groupby_ = super(SaleReport, self)._group_by_pos(groupby=groupby)
        groupby_ += ", pos.new_field_id"
        return groupby_
```

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by celm1990** [comment]
@pimodoo @mart-e sorry for ping can you help me with this PR please

**Comment by mart-e** [comment]
@celm1990 somebody from @odoo/rd-sale should have a look but you can already see that you have some issues with the ci/style
https://runbot.odoo.com/runbot/build/9331755
Solving these would be a good start :)

**Comment by celm1990** [comment]
> @celm1990 somebody from @odoo/rd-sale should have a look but you can already see that you have some issues with the ci/style
> https://runbot.odoo.com/runbot/build/9331755
> Solving these would be a good start :)

Hi @mart-e thanks for your reply, i fixed ci/style and all checks are ok. I wait for review i  hope can be merge before release V15.

File: addons/pos_sale/report/sale_report.py
Comment (Feyensv) [inline addons/pos_sale/report/sale_report.py:67]:
```suggestion
    def _from_pos(self):
```

The from_clause is defined from the sale POV.  It's not used here and therefore shouldn't be given as parameter to this method.

File: addons/pos_sale/report/sale_report.py
Comment (Feyensv) [inline addons/pos_sale/report/sale_report.py:81]:
```suggestion
    def _group_by_pos(self):
```
Same comment than for `_from_pos`

File: addons/pos_sale/report/sale_report.py
Comment (Feyensv) [inline addons/pos_sale/report/sale_report.py:109]:
```suggestion
        res = super()._query(with_clause, fields, groupby, from_clause)
```
Specifying the class is not necessary ;)

**Review by Feyensv — COMMENTED** [review]
Some little remarks, but globally LGTM :)

Also, I believe the word is "facilitate" and not "facilite" (for your commit & PR titles), but "ease" also works :D

**Comment by celm1990** [comment]
> Some little remarks, but globally LGTM :)
> 
> Also, I believe the word is "facilitate" and not "facilite" (for your commit & PR titles), but "ease" also works :D

Hi @Feyensv I apply your recommendations i hope all ok, thanks for review, sorry for bad english (:

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Comment by celm1990** [comment]
@Feyensv only curiosity, is possible merge into V12 and forwardport to V13, V14 too???

**Comment by Feyensv** [comment]
I'm not strictly opposed to these kind of changes in stable versions, but I still prefer to merge them in master.

This diff looks safe, but I've seen enough bugs created from "clean and safe" bugfixes in stable versions, sometimes in third party modules.  Therefore, unless you really need it for a clear use-case in stable, I'd prefer to keep it for master :).

**Pull request closed without merging**
